fastplotlib/layouts/graphic_methods_mixin.py

```diff
@@ -0,0 +1,558 @@
+from typing import *
+
+import numpy
+import weakref
+
+from ..graphics import *
+
+
+class GraphicMethodsMixin:
+    def __init__(self):
+        pass
+
+    def _create_graphic(self, graphic_class, *args, **kwargs) -> weakref.proxy:
+        if 'center' in kwargs.keys():
+            center = kwargs.pop('center')
+        else:
+            center = False
+
+        if 'name' in kwargs.keys():
+            self._check_graphic_name_exists(kwargs['name'])
+
+        graphic = graphic_class(*args, **kwargs)
+        self.add_graphic(graphic, center=center)
+
+        # only return a proxy to the real graphic
+        return weakref.proxy(graphic)
+
+    def add_heatmap(self, data: Any, vmin: int = None, vmax: int = None, cmap: str = 'plasma', filter: str = 'nearest', chunk_size: int = 8192, isolated_buffer: bool = True, *args, **kwargs) -> weakref.proxy(HeatmapGraphic):
```

Comment:
can you change the type annotations to use `[]` instead of `()`:

```python
-> weakref.proxy[Graphic]
```

Reply:
if I change the type annotation to this, it thinks that I am trying to index `weakref.proxy`